File: app/src/main/java/com/timsimonhughes/atlas/network/NasaPotdService.java
package com.timsimonhughes.atlas.network;

import com.timsimonhughes.atlas.model.POTD;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface NasaPotdService {

    @GET("planetary/apod")
    Call<POTD> getPhotoOfDay(@Query("api_key") String apiKey);


    @GET("planetary/apod")
    Call<POTD> getPOTDByDate(@Query("date") String date, @Query("api_key") String apiKey);

    @GET("planetary/apod")
    Call<List<POTD>> getPOTDByDateRange(@Query("start_date") String startDate, @Query("end_date") String endDate, @Query("api_key") String apiKey);

//    @GET("api/v1/rover/curiosity../mars-photos")
//    Call<List>


    ///////// PHOTO OF THE DAY EXAMPLE ENDPOINTS /////////
    // https://api.nasa.gov/planetary/apod?date=2019-02-01&api_key=pl521eBJWqAwQ8rIOpleax8WwD9e2aXaxhCpjw0U - SPECIFIC DATE QUERY
    // https://api.nasa.gov/planetary/apod?start_date=2018-01-01&end_date=2018-01-03&api_key=pl521eBJWqAwQ8rIOpleax8WwD9e2aXaxhCpjw0U - DATE RANGE QUERY


}

File: app/src/main/java/com/timsimonhughes/atlas/network/NasaPotdService.kt
package com.timsimonhughes.atlas.network

import com.timsimonhughes.atlas.model.POTD
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface NasaPotdService {

    @GET("planetary/apod")
    fun getPhotoOfDay(@Query("api_key") apiKey: String): Call<POTD>

    @GET("planetary/apod")
    fun getPOTDByDate(
            @Query("date") date: String,
            @Query("api_key") apiKey: String): Call<POTD>

    @GET("planetary/apod")
    fun getPOTDByDateRange(
            @Query("start_date") startDate: String,
            @Query("end_date") endDate: String,
            @Query("api_key") apiKey: String): Call<List<POTD>>

}
